src/app/zap-templates/templates/app/cluster-objects.zapt

```diff
@@ -35,29 +35,33 @@ namespace {{name}} {
     public:
         {{#zcl_struct_items}}
         {{#if isArray}}
-        DataModel::List<{{#if_is_enum type}}{{type}}{{else}}{{asChipZapType type}}{{/if_is_enum}}> {{asLowerCamelCase label}};
+        DataModel::List<{{#if_is_enum type}}{{type}}{{else}}{{asChipZapType type false}}{{/if_is_enum}}> {{asLowerCamelCase label}};
         {{else}}
-        {{#if_is_enum type}}{{type}}{{else}}{{asChipZapType type}}{{/if_is_enum}} {{asLowerCamelCase label}};
+        {{#if_is_enum type}}{{type}}{{else}}{{asChipZapType type false}}{{/if_is_enum}} {{asLowerCamelCase label}};
         {{/if}}
         {{/zcl_struct_items}}
 
         CHIP_ERROR Encode(TLV::TLVWriter &writer, uint64_t tag) const;
+        {{#if struct_contains_array}}
+        {{else}}
         CHIP_ERROR Decode(TLV::TLVReader &reader);
+        {{/if}}
     };
 
     {{#if struct_contains_array}}
     struct DecodableType {
     public:
         {{#zcl_struct_items checkForDoubleNestedArray="true"}}
         {{#if isArray}}
-        DataModel::DecodableList<{{#if struct_item_contains_nested_array}}{{type}}::DecodableType{{else}}{{#if_is_enum type}}{{type}}{{else}}{{asChipZapType type}}{{/if_is_enum}}{{/if}}> {{asLowerCamelCase label}};
```

Comment:
After this the `struct_item_contains_nested_array` helper is unused.  Should file a followup on ZAP to consider removing it, unless we expect we'll need it for something else.